source/Octopus.Tentacle/Commands/RunAgentCommand.cs

```diff
@@ -167,10 +169,16 @@ protected override void Stop()
                 workspaceCleanerTask.Value.Stop();
             }
 
-            if (kubernetesPodMonitorHasStarted)
+            if (kubernetesPodMonitorTask.IsValueCreated)
             {
                 kubernetesPodMonitorTask.Value.Stop();
```

Comment:
If start hasn't been called, Stop is a noop. We can just check if the Value has been created and if it has, call Stop no matter what. We really just want to avoid creating the instance at Stop time if it's not already created.